packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1442,7 +1442,7 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
       _showCaretOnScreen();
       if (!_value.selection.isValid) {
         // Place cursor at the end if the selection is invalid when we receive focus.
-        widget.controller.selection = TextSelection.collapsed(offset: _value.text.length);
+        _handleSelectionChanged(TextSelection.collapsed(offset: _value.text.length), renderEditable, SelectionChangedCause.tap);
```

Comment:
Fix 2/2: This fixes the case where an autofocused field couldn't ever display its selection menu.  The reason was that the menu was never created.

However, this had an implication in our tests.  I had to wrap a bunch of pumped test widgets in `MaterialApp`, because otherwise I was getting an error that there was no parent `Overlay` widget when it tried to create the selection menu.  See the changes below.

Reply:
Also, I just remembered that I used `SelectionChangedCause.tap` and it felt a bit hacky.  This isn't isn't caused by a tap.  Really, it's called when the widget is rebuilt and is automatically focused.  Should I add a new value to the `SelectionChangedCause` enum?

Reply:
I feel a bit hacky too, but I can't think of a good cause either. I wonder if it make sense to leave it null?

Reply:
Good idea.  I thought it couldn't be null for some reason, but it seems to work fine.